cron-ci/build-hpc-stack.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash -l
+# pass -l (treat script as login shell) so lmod module environment is picked up
+set -eux
+
+# Parse hpc-stack yaml file to get version variables
+source stack_helpers.sh
+eval $(parse_yaml ${HPC_STACK_FILE} "STACK_")
+
+# get the current hpc-stack hash and compare it to the previously built hash (if it exists)
+# if it exists and matches current hash, don't build
+current_hash=$(git rev-parse HEAD)
+
+build=true
+if [[ -f "${HPC_HOMEDIR}/prev_hash.txt" ]]; then
+    prev_hash=$(cat ${HPC_HOMEDIR}/prev_hash.txt)
+    if [[ "$current_hash" == "$prev_hash" ]]; then
+        build=false
+        echo `date`
+        echo ""
+        echo "hpc-stack version has not changed since last time. Not building."
+        echo "hpc-stack hash: ${current_hash}"
+    fi
+fi
+
+# Return 42 if not built because hash value hasn't changed
+[[ "$build" == false ]] && ( echo "Nothing has changed, so nothing to build"; exit 42)
```

Comment:
why 42?
This is not an error.  Non-zero return codes are treated as errors.

Reply:
Do you have a better suggestion of a way to indicate to the parent script that hpc-stack wasn't built, as to not run the tests?

Or just run the tests anyway?

Reply:
I see what you mean.

Reply:
I think the simple answer is to just move that check up to the top-level script.